modules/apps/portal/portal-log4j-test/src/test/java/com/liferay/portal/log4j/Log4jTest.java

```diff
@@ -0,0 +1,464 @@
+
+		if (_tempFile != null) {
+			_tempFile.delete();
+		}
+
+		File logDir = new File(
+			StringUtil.replace(System.getProperty("user.dir"), '\\', '/'),
+			"logs");
+
+		for (File file : logDir.listFiles()) {
+			file.delete();
+		}
+
+		logDir.delete();
+	}
+
+	@Test
+	public void testConsoleAppender() {
+		Log log = LogFactoryUtil.getLog(Log4jTest.class);
+
+		log.info("Test Message");
+
+		String[] logMessages = StringUtil.splitLines(_baos.toString());
+
+		String expectedOutput = logMessages[logMessages.length - 1];
+
+		_assertLog(expectedOutput, 147);
+	}
+
+	@Test
+	public void testGetOriginalLevel() {
+		String level = Log4JUtil.getOriginalLevel(_NAMES[5]);
+
+		Assert.assertEquals("The original level should be INFO", level, "INFO");
+
+		level = Log4JUtil.getOriginalLevel(_NAMES[6]);
+
+		Assert.assertEquals(
+			"The original level should be DEBUG", level, "DEBUG");
+	}
+
+	@Test
+	public void testLoggerEnabled() {
+		Log log = LogFactoryUtil.getLog(_NAMES[0]);
+
+		Assert.assertTrue("Logger should be all enabled", log.isTraceEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[1]);
+
+		Assert.assertFalse(
+			"Setting logger level OFF does not take effect",
+			log.isTraceEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[2]);
+
+		Assert.assertTrue(
+			"Setting logger level FATAL does not take effect",
+			log.isFatalEnabled() && !log.isErrorEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[3]);
+
+		Assert.assertTrue(
+			"Setting logger level ERROR does not take effect",
+			log.isErrorEnabled() && !log.isWarnEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[4]);
+
+		Assert.assertTrue(
+			"Setting logger level WARN does not take effect",
+			log.isWarnEnabled() && !log.isInfoEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[5]);
+
+		Assert.assertTrue(
+			"Setting logger level INFO does not take effect",
+			log.isInfoEnabled() && !log.isDebugEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[6]);
+
+		Assert.assertTrue(
+			"Setting logger level DEBUG does not take effect",
+			log.isDebugEnabled() && !log.isTraceEnabled());
+
+		log = LogFactoryUtil.getLog(_NAMES[7]);
+
+		Assert.assertTrue(
+			"Setting logger level TRACE does not take effect",
+			log.isTraceEnabled());
+	}
+
+	@Test
+	public void testRollingFileAppender() throws Exception {
+		Log log = LogFactoryUtil.getLog(Log4jTest.class);
+
+		log.info("Test Message");
+
+		File logDir = new File(
+			StringUtil.replace(System.getProperty("user.dir"), '\\', '/'),
+			"logs");
+
+		String content = "";
+
+		Matcher matcher = null;
+
+		for (File file : logDir.listFiles()) {
+			String fileName = file.getName();
+
+			URI uri = file.toURI();
+
+			URL url = uri.toURL();
+
+			if (fileName.endsWith(".log")) {
+				matcher = _textFileNamePattern.matcher(fileName);
+
+				Assert.assertTrue(
+					"test file name should be " + fileName, matcher.matches());
+
+				try (InputStream inputStream = url.openStream()) {
+					byte[] bytes = _getBytes(inputStream);
```

Comment:
I changed the file reading here with `petra.io.StreamUtil`. After a second though, maybe it doesn't need to be this. If you want lines from the file, use JDK API `Files.readFully`.